dandi/download.py

```diff
@@ -1141,8 +1141,12 @@ def _check_if_more_attempts_allowed(
                 exc,
             )
         else:
-            lgr.debug("%s - download failed: %s", path, exc)
-            return None
+            lgr.debug(
+                "%s - download failed on attempt #%d: %s, will sleep a bit and retry",
+                path,
+                attempt,
+                exc,
+            )
     elif attempt >= attempts_allowed:
```

Comment:
@yarikoptic Question: Shouldn't this check be at the top of the function?  Otherwise, if we keep hitting, say, 500 responses, it'll keep retrying beyong the maximum number of attempts.

Reply:
indeed. But **after** potential increment of `attempts_allowed` in the "aggressive" mode.

Reply:
Fixed in bb05a23f.